bigframes/session/__init__.py

```diff
@@ -307,6 +309,72 @@ def read_gbq(
                 api_name="read_gbq",
             )
 
+    def _filters_to_query(self, query_or_table, filters):
+        """Convert filters to query"""
+
+        if (filters is None) or (len(filters) == 0):
+            return query_or_table
+
+        valid_operators = ["IN", "NOT IN", "=", ">", "<", ">=", "<=", "!="]
+
+        sub_query = (
+            f"({query_or_table})" if _is_query(query_or_table) else query_or_table
+        )
+
+        where_clause = ""
+        if filters:
+            if not isinstance(filters, list):
+                raise ValueError("Filters should be a list.")
+
+            if not (
+                all(isinstance(item, list) for item in filters)
+                or all(isinstance(item, tuple) for item in filters)
+            ):
+                raise ValueError(
+                    "All items in filters should be either all lists or all tuples."
+                )
+
+            if all(isinstance(sub_filter, tuple) for sub_filter in filters):
+                filters = [filters]
+
+            grouped_expressions = []
+            for group in filters:
+                if not isinstance(group, list):
+                    raise ValueError("Each filter group should be a list.")
+
+                group_expressions = []
+                for filter_item in group:
+                    if not isinstance(filter_item, tuple):
+                        raise ValueError("Each filter condition should be a tuple.")
+
+                    column, operator, value = filter_item
+                    operator = operator.upper()
+
+                    if operator not in valid_operators:
+                        raise ValueError(f"Operator {operator} is not valid.")
+
+                    if operator in ["IN", "NOT IN"]:
+                        if not isinstance(value, list):
+                            raise ValueError(
+                                f"Value for operator {operator} should be a list."
+                            )
```

Comment:
We don't need this check. Any iterable should be OK.

Reply:
Done.